### DISCUSSION TITLE
When I go back, it goes back to the first page

### DISCUSSION THREAD

**Discussion by hugopvl** [directus/directus#9151] [Feature Requests]
Good morning guys,

An issue that I think improved navigation.

In a case where we have hundreds of items, with several pages, I'm on page 20 of 50, I open an item to consult, when I go back, it goes back to the first page, shouldn't I keep it on page 20 to continue the search?

**Comment by benhaynes** [comment]
Yes, I believe I opened an issue about this. Definitely needs a patch.

**Reply by rijkvanzanten** [reply]
This raises an interesting question, that's similar to the "when to reset scroll to 0 and when not" question (cc @joselcvarela) as we'd want to persist the Page when going to/from a detail page, but _not_ when you switch collections, or change sort/filter or anything like that 🤔 We should figure out what those exact rules are

**Reply by joselcvarela** [reply]
I already started a draft PR with the scroll position https://github.com/directus/directus/pull/9146. But this goes even further.
Maybe we need to persist a state before navigate and check what need to be applied

**Reply by rijkvanzanten** [reply]
Hmm yeah, though that feels like quite a bit of overhead to add dedicated rules per page. I'm wondering if there's a (set of) rule(s) we can come up with that'll work everywhere. One obvious one is that we don't want to scroll when the route is a _subroute_ of another route, as we use those for drawers 🤔

**Comment by br41nslug** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
